articles/capsule-networks-an enhancement-to-convolution-neural-networks/index.md

```diff
@@ -0,0 +1,195 @@
+In CNN's, we have several pooling layers. Researchers have found that these pooling layers tend to lose information, such as the objects' precise location and pose. Capsule Networks are Equivariance, making them suitable for applications such as object detection and image segmentation.
+
+CapsNet has four main components:
+
+1. Matrix Multiplication.
+
+This operation is performed on the input layer. We take an image that we want to convert into vector values to understand the spatial layout.
+
+2. Scalar weighting of the inputs
+
+The inputs' weighting helps determine the direction where the current capsules should go in the next layer. It works alongside the third component mentioned below. 
+
+3. Dynamic Routing Algorithm
+
+The role of this algorithm is to facilitate communication between the Primary Capsule and the DigitCaps layer. 
+
+4. Applying a non-linear function to condense the information
+
+We use the squashing (non-linear) function shown below to condense the information to a value between 0 and 1. The short vectors are squashed to an almost zero value, while the long vectors are squashed to a length of slightly less than one. Thus, the length of the output vector of a capsule denotes the probability of whether an entity is present in the current input.
+Instead of adding the squashing function to each layer as in CNNs, the function is applied to a nested set of layers in a CapsNet.
+
+It is important to note that the squashing function preserves the vectors' orientation, but it squashes it to ensure its length is between 0 and 1. 
+
+The [paper](https://www.cs.toronto.edu/~hinton/absps/DynamicRouting.pdf/) introduces a new squashing (non-linearity) function shown below, which works best with capsules:
+
+![The squashing function used in ConvNets](/engineering-education/capsule-networks-an-enhancement-to-convolution-neural-networks/squashing-function.PNG)<br>
+
+*[Image Source: University of Toronto](https://web.cs.toronto.edu/)*
+
+The term vj refers to the output after applying the squashing function; the term sj refers to the previous step's output.
+
+### The CapsNet Architecture
+
+The capsule network architecture contains an encoder and a decoder, with each layer consisting of 3 layers.
+
+![A simple CapsNet with 3 layers](/engineering-education/capsule-networks-an-enhancement-to-convolution-neural-networks/caps-net-3-layers.PNG)<br>
+
+*[Image Source: University of Toronto](https://web.cs.toronto.edu/)*
+
+The encoder contains three layers:
+
+1. Convolution layer (Conv1)
+
+This layer has 256 convolutional kernels with 9 x 9 that slide with a stride of 1. The ReLU activation function is also used. This layer is responsible for converting pixel intensities to the activities of local feature detectors. These outputs are then fed as inputs to the primary capsule layer.
+
+2. Primary Capsule layer
```

Comment:
You talk about inverse rendering. Let's say I am performing an object detection task. How does the network learn the task? First, the image goes through the network, and later it learns about where the object is? The doubt is that since the input is the image and the output is the object's coordinates, isn't that happening already in a CNN? There are low dimensional features extracted in earlier layers and higher-level features in the later layers. It is not quite clear how inverse rendering occurs.